logstash-core/lib/logstash/webserver.rb

```diff
@@ -20,7 +20,8 @@ def initialize(logger, options={})
       http_port    = options[:http_port] || 9600
       @options     = {}
       @cli_options = options.merge({ :rackup => ::File.join(::File.dirname(__FILE__), "api", "init.ru"),
-                                     :binds => ["tcp://#{http_host}:#{http_port}"] })
+                                     :binds => ["tcp://#{http_host}:#{http_port}"],
+                                     :debug => logger.debug? })
```

Comment:
Is necessary to tell puma he is suposed to work on debug mode, only
possible throught cli options.

On Fri, 12 Feb 2016 13:12 João Duarte <notifications@github.com> wrote:

> In logstash-core/lib/logstash/webserver.rb
> <https://github.com/elastic/logstash/pull/4681#discussion_r52734283>:
>
> > @@ -20,7 +20,8 @@ def initialize(logger, options={})
> >        http_port    = options[:http_port] || 9600
> >        @options     = {}
> >        @cli_options = options.merge({ :rackup => ::File.join(::File.dirname(__FILE__), "api", "init.ru"),
> > -                                     :binds => ["tcp://#{http_host}:#{http_port}"] })
> > +                                     :binds => ["tcp://#{http_host}:#{http_port}"],
> > +                                     :debug => logger.debug? })
>
> is this still necessary? isn't the logger passed to the webserver?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4681/files#r52734283>.
>